#! /usr/bin/env python
# bta_tof_driver configuration

PACKAGE='bta_tof_driver'
import roslib; roslib.load_manifest(PACKAGE)


from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

#Note: Reconfiguration level is currently set to 0 for all the parameters, assuming that the parameters can be changed when when the sensor is running....

#       Name			Type      	Reconfiguration level   Description	Default    	Min   	Max

gen.add("Integration_Time",	int_t,	0, "Integration time(in msec) for the sensor",1500,	100,25000)

gen.add("Frame_rate", double_t,0,"Select the camera frame rate in frame per second",15.0,1.0,160.0)

gen.add("Read_reg", bool_t, 0, "Read register", False)
gen.add("Reg_addr", str_t,0,"Register address", "0x0")
gen.add("Write_reg", bool_t, 0,"Write register", False)
gen.add("Reg_val", str_t,0,"Register value", "0x0")


exit(gen.generate(PACKAGE, "bta_tof_driver", "bta_tof_driver"))

